### PR TITLE
add new control-disablements sub-module

### PR DESCRIPTION
**Pull request by mcalhoun**
## what

* Add a new control disablements module that provides a list of controls to be disabled based on the input variables.

## why

* Certain controls should only be enabled in one region per account (the global collection region) while others should only be enabled in the central CloudTrail account.

### CODE REVIEW

File: modules/control-disablements/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#-----------------------------------------------------------------------------------------------------------------------
+data "aws_caller_identity" "this" {}
+data "aws_region" "this" {}
+
+locals {
+  is_global_resource_region     = data.aws_region.this.name == var.global_resource_collector_region
+  is_central_cloudtrail_account = data.aws_caller_identity.this.account_id == var.central_logging_account
+
+  all_global_region_controls = [
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/aws-foundational-security-best-practices/v/1.0.0/Config.1",
```
Comment (osterman) [inline modules/control-disablements/main.tf]: We should use the AWS partition here no?

File: modules/control-disablements/main.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.6",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.7",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.8",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.9",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.10",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.11",
+    "arn:aws:securityhub:${data.aws_region.this.name}:${data.aws_caller_identity.this.account_id}:control/cis-aws-foundations-benchmark/v/1.2.0/1.12",
```
Comment (nitrocode) [inline modules/control-disablements/main.tf]: Saving the prefix to a `local` would be good

```suggestion
    "${local.arn_prefix}:control/cis-aws-foundations-benchmark/v/1.2.0/1.12",
```

File: modules/control-disablements/main.tf (hunk 1 above)
Comment (mcalhoun) [reply modules/control-disablements/main.tf]: 12d28d1b9a60a8c590d8074c6983d4358624c3b4

File: modules/control-disablements/main.tf (hunk 2 above)
Comment (mcalhoun) [reply modules/control-disablements/main.tf]: 71b67bd881105fd44184e4ee81eb0727c0a0081a

**Review by osterman — DISMISSED** [review]

File: modules/control-disablements/main.tf [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.11",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.12",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.13",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.14",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.16",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.20",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/1.22",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/2.5",
+  ]
+  included_global_region_controls = [for c in local.all_global_region_controls : c if ! local.is_global_resource_region]
```
Comment (nitrocode) [inline modules/control-disablements/main.tf]: Simpler expression

```suggestion
  included_global_region_controls = local.is_global_resource_region ? local.all_global_region_controls : []
```

File: modules/control-disablements/main.tf [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.7",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.8",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.9",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.10",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.11",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.12",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.13",
+    "${local.control_prefix}/cis-aws-foundations-benchmark/v/1.2.0/3.14",
+  ]
+  included_cloudtrail_controls = [for r in local.all_cloudtrail_controls : r if ! local.is_central_cloudtrail_account]
```
Comment (nitrocode) [inline modules/control-disablements/main.tf]: Simpler expression

```suggestion
  included_cloudtrail_controls = local.is_central_cloudtrail_account ? local.all_cloudtrail_controls : []
```

File: modules/control-disablements/main.tf (hunk 3 above)
Comment (mcalhoun) [reply modules/control-disablements/main.tf]: 0093448fc66ead94b1bd2f2f8f9fbc3a3822a9ee

File: modules/control-disablements/main.tf (hunk 4 above)
Comment (mcalhoun) [reply modules/control-disablements/main.tf]: 0093448fc66ead94b1bd2f2f8f9fbc3a3822a9ee

**Review by nitrocode — APPROVED** [review]
LGTM

**Pull request merged**